### ISSUE TITLE
Replication "isSensitive" logic prevents "boss" from creating any report for users directly under them in hierarchy

### ISSUE THREAD

**Issue by kennsippell** [medic/cht-core#6370]
**To Reproduce**
1. Take a hierarchy with places `A > B`
1. User 1 has A as their facility_id
1. User 2 has B as their facility_id
1. User 1 logs in, goes to reports tab, creates a report with place_id=B
1. User 2 logs in, and expects to see that report

**Expected behavior**
In this case, A is not B's "boss" and is not passing sensitive data. This isn't a report about B, it is just a report that B needs to see.

**Logs**
If applicable, include the server or browser logs.

**Screenshots**

**Environment**
- Instance: msf-niger.dev.medicmobile.org
- App: webapp
- Version: master

**Additional context**
Seems this is due to the `isSensitive` logic [here](https://github.com/medic/cht-core/blob/master/api/src/services/authorization.js#L330)

Perhaps this can be disabled via configuration? Or perhaps forms can opt-in to this experience via a flag.

**Comment by kennsippell** [comment]
Adding COVID-19 flag as this bug is impacting msf-niger project. I'm working around it by having random meaningless elements in the hierarchy, but it is awkward and we all know how hard it is the change a hierarchy for a production system.

**Comment by SCdF** [comment]
For reference, this logic was introduced here: https://github.com/medic/cht-core/issues/2648

**Comment by abbyad** [comment]
May be worth noting that #6352 has a related requirement for a different use case.

**Comment by kennsippell** [comment]
We've worked around this by adding unnecessary places into the hierarchy called "filing cabinets" instead of using a custom build.  It's pretty hard to change a hierarchy, so not sure if this is needed on a COVID timeframe.

**Comment by garethbowen** [comment]
There is also a range of issues related to sensitive data where only certain users or roles should be able to access certain reports for PHI reasons. The ideal solution would take this into account rather than blocking all reports about your place that were filed by someone above you. In the short term we could just add a configuration to enable or disable this feature completely which would unblock these configurations and give us more time to come up with the ideal solution.

**Comment by garethbowen** [comment]
Added to 3.10.0 for consideration.

**Comment by kennsippell** [comment]
Turns out this is an inconvenience for two user-types. One of which needs to actually make new contacts. You seem supportive of pursuing an ideal solution in the near-term, and it really does keep things a lot cleaner. The current config-msf/niger code relies on the workaround.

**Comment by garethbowen** [comment]
That would be a quick fix, but I think we need to revisit the idea of `sensitive` reports.

Usually only certain reports will be sensitive, such as a supervisor submitting a performance report about the CHP. I think a better solution would be to have a configuration on forms about whether the information should be considered sensitive (we may need to come up with a better term). I think this should default to "false". This would be a breaking change however I think nobody is intentionally using this feature currently, so we can probably get away with listing it as a breaking change in the release notes. If the form were deemed "sensitive" then the current logic would kick in.

There is another concept of "sensitive" which is for information such as HIV status that should only be available to very specific users. This differs from the existing concept because it's not information about the current user. For this I would think about having a role (or array of roles) configured on the form so, for example, an HIV Nurse can see reports about HIV status. I think this other definition of separate differs enough from the other definition that it should be raised and developed as a separate feature.

**Comment by dianabarsan** [comment]
Oh, a turnaround, I like! 

So the new isSensitive is: 
- report has `sensitive = true` and ( is about you or your place )
- report has `sensitive = Array` and (`sensitive` does not include your roles)

As for renaming, looking here https://en.wikipedia.org/wiki/Information_sensitivity there are a couple of adjectives used: *sensitive*, *confidential*, *private* and *classified*.

**Comment by garethbowen** [comment]
I wonder if we should use two different words for the two different types, eg: `private` means don't show the subject, and `sensitive` means special roles required. Feel free to just to the `private` bit now, so we can take the time to do the `sensitive` bit right.

**Comment by dianabarsan** [comment]
Ready for AT on `6370-private-reports`. [Documentation PR](https://github.com/medic/cht-docs/pull/143)

**Comment by brad1905** [comment]
LGTM

**Comment by dianabarsan** [comment]
Merged into `master`.
